java-checks-test-sources/src/main/files/non-compiling/checks/TextBlocksInComplexExpressionsCheck.java

```diff
@@ -0,0 +1,74 @@
+package test;
+
+class TextBlocksInComplexExpressionsCheck {
+  
+  void fun() {
+    listOfString.stream()
+      
+      .map(str -> { // Noncompliant@+1 [[sc=18;ec=8]]{{Refactor this text block to a local variable or a static final field.}}
```

Comment:
When the issue is on several lines we can also test the end line: `[[sc=18;el=+14;ec=8]]`